airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -124,32 +129,39 @@ def streams(self, config: Mapping[str, Any]) -> List[Type[Stream]]:
             config.start_date = validate_start_date(config.start_date)
             config.end_date = validate_end_date(config.start_date, config.end_date)
 
-        api = API(account_id=config.account_id, access_token=config.access_token, page_size=config.page_size)
+        api = API(account_ids=config.account_ids, access_token=config.access_token, page_size=config.page_size)
 
         # if start_date not specified then set default start_date for report streams to 2 years ago
         report_start_date = config.start_date or pendulum.now().add(years=-2)
 
         insights_args = dict(
-            api=api, start_date=report_start_date, end_date=config.end_date, insights_lookback_window=config.insights_lookback_window
+            api=api,
+            account_ids=config.account_ids,
+            start_date=report_start_date,
+            end_date=config.end_date,
+            insights_lookback_window=config.insights_lookback_window,
         )
         streams = [
-            AdAccount(api=api),
+            AdAccount(api=api, account_ids=config.account_ids),
             AdSets(
                 api=api,
+                account_ids=config.account_ids,
```

Comment:
I deleted `account_ids` from the API class, as it doesn't need it. Also, getting `account_ids` from API would break the single responsibility principle. Thanks for noticing!